runhouse/rns/blobs/blob.py

```diff
@@ -0,0 +1,291 @@
+
+        path = str(
+            path or self.default_path(self.rns_address, system)
+        )  # Make sure it's a string and not a Path
+
+        from runhouse.rns.blobs.file import File
+
+        new_blob = File(path=path, system=system, data_config=data_config)
+        new_blob.write(self.fetch())
+        return new_blob
+
+    def fetch(self):
+        """Return the data for the user to deserialize.
+
+        Example:
+            >>> data = blob.fetch()
+        """
+        if not self.system:
+            return self._data
+        if self.system.on_this_cluster():
+            return obj_store.get(self.name)
+        return self.system.get(self.name)
+
+    def _save_sub_resources(self):
+        if isinstance(self.system, Resource):
+            self.system.save()
+
+    def rename(self, name: str):
+        """Rename the blob.
+
+        Example:
+            >>> blob = rh.blob(data)
+            >>> blob.rename("new_name")
+        """
+        if self.name == name:
+            return
+        old_name = self.name
+        self.name = name  # Goes through Resource setter to parse name properly (e.g. if rns path)
+        # Also check that this is a Blob and not a File
+        if isinstance(self.system, Cluster) and self.__class__.__name__ == "Blob":
+            if self.system.on_this_cluster():
+                obj_store.rename(old_key=old_name, new_key=self.name)
+            else:
+                self.system.rename(old_key=old_name, new_key=self.name)
+
+    def save(
+        self,
+        name: str = None,
+        overwrite: bool = True,
+    ):
+        # Need to override Resource's save to handle key changes in the obj store
+        # Also check that this is a Blob and not a File
+        if name and not self.name == name and self.__class__.__name__ == "Blob":
+            if overwrite:
+                self.rename(name)
+            else:
+                if isinstance(self.system, Cluster):
+                    if self.system.on_this_cluster():
+                        obj_store.put(self.name, self.fetch())
+                    else:
+                        self.system.put(name, self.fetch())
+        super().save(name=name, overwrite=overwrite)
+
+    def write(self, data):
+        """Save the underlying blob to its cluster's store.
+
+        Example:
+            >>> rh.blob(data).write()
+        """
+        if not self.system:
+            self._data = data
+        elif self.system.on_this_cluster():
+            obj_store.put(self.name, data)
+        else:
+            self.system.put(self.name, data)
+        return self
+
+    def rm(self):
+        """Delete the blob from wherever it's stored.
+
+        Example:
+            >>> blob = rh.blob(data)
+            >>> blob.rm()
+        """
+        if self.system is None:
+            self._data = None
+        if self.system.on_this_cluster():
+            obj_store.delete(self.name)
+        else:
+            self.system.delete(self.name)
```

Comment:
Is this supposed to delete the blob from the cluster's local file system? We don't have a `delete` method for the cluster which does this (i think previously this went through the blob's `folderz, how can we do this now that a blob doesn't live in a folder object?